python-scripts/xbox/interface/if_xbdm.py

```diff
@@ -77,21 +81,32 @@ def xbdm_command(cmd, length=None):
   return lines
 
 def GetModules():
-  return [] # FIXME: Make this work..
+  modulesList = []
   lines = xbdm_command("modules")
-  for line in lines:
-    print(line)
-    chunks = line.split()
-    for chunk in chunks:
-      if (chunk[0:6] == b'name=\"'):
-        print("  name = '" + str(chunk[6:-1]) + "'")
-      elif (chunk[0:5] == b'base='):
-        print("  base = '" + str(chunk[5]) + "'")
-      else:
-        print('  unparsed: ' + str(chunk))
-        pass
-        #assert(False)
-  return []
+
+  linesList = lines
+
+  for line in linesList: 
+
+    nameAddrDict ={}
+    line = line.decode("utf-8") 
+
+    nameAddrChunks = line.split()
+
+    for nameAddrChunk in nameAddrChunks:
+       singleChunk = nameAddrChunk.split("=")    #name="xbdm.dll" ->  ['name', '"xbdm.dll"']
+
+       if(singleChunk[0] =="name"):  
+        nameAddrDict[singleChunk[0]]= singleChunk[1].strip("\"")  #name of the module (with out >"< (Quote))
+                                                
+       elif(len(singleChunk)==1):      
+         nameAddrDict[singleChunk[0]]= "true" 
+
+       else:
+         nameAddrDict[singleChunk[0]]=int(singleChunk[1], 0) 
+
+    modulesList.append(nameAddrDict)
+  return modulesList
```

Comment:
The commented lines following this can be removed or they should be moved inside this function (with a comment saying that it's example of a response)